src/compiler/transformers/es2015.ts

```diff
@@ -2075,7 +2075,7 @@ namespace ts {
                     const firstDeclaration = firstOrUndefined(declarations);
                     if (firstDeclaration) {
                         const lastDeclaration = lastOrUndefined(declarations);
-                        setSourceMapRange(declarationList, createRange(firstDeclaration.pos, lastDeclaration.end));
+                        setSourceMapRange(declarationList, createRange(firstDeclaration.pos, lastDeclaration.end, /*noAssert*/ true)); // TODO: GH#23370

```

Comment:
We sometimes insert temp variables into the declaration list, and the first temp we insert caches the initializer on the RHS. Since all we care about is inserting a mapping at the beginning and end of the list, we could probably just change this to:

```ts
if (declarations.length) {
    const range = reduceLeft(declarations, (range, node) => {
        range.pos = range.pos === -1 ? node.pos : node.pos === -1 ? range.pos : Math.min(range.pos, node.pos);
        range.end = range.end === -1 ? node.end : node.end === -1 ? range.end : Math.max(range.end, node.end);
        return range;
    }, createRange(-1, -1));
    setSourceMapRange(declarationList, range);
}
```

Though we would need to verify the results in the debugger.